examples/launch_contributed_surrogate_bo.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import copy
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+from sklearn.linear_model import BayesianRidge
+
+from syne_tune import Tuner, StoppingCriterion
+from syne_tune.backend import LocalBackend
+from syne_tune.config_space import randint
+from syne_tune.optimizer.schedulers import FIFOScheduler
+from syne_tune.optimizer.schedulers.searchers.bayesopt.sklearn.contributed_surrogate_searcher import (
+    ContributedSurrogateSearcher,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.sklearn.estimator import (
+    SklearnEstimator,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.sklearn.predictor import (
+    SklearnPredictor,
+)
+from syne_tune.optimizer.schedulers.searchers.bayesopt.tuning_algorithms.bo_algorithm_components import (
+    IndependentThompsonSampling,
+)
+
+
+class BayesianRidgePredictor(SklearnPredictor):
+    """
+    Base class for the sklearn predictors
+    """
+
+    def __init__(self, ridge: BayesianRidge):
+        self.ridge = ridge
+
+    def predict(
+        self, X: np.ndarray, return_std: bool = True
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Returns signals which are statistics of the predictive distribution at
+        input points ``inputs``.
+
+
+        :param inputs: Input points, shape ``(n, d)``
+        :return: Tuple with the following entries:
+            * "mean": Predictive means in shape of ``(n,)``
+            * "std": Predictive stddevs, shape ``(n,)``
+        """
+        print(f"Predicting with BayesianRidgePredictor using X.shape={X.shape}")
+        return self.ridge.predict(X, return_std=True)
+
+
+class BayesianRidgeEstimator(SklearnEstimator):
+    """
+    Base class for the sklearn Estimators
```

Comment:
`SklearnEstimator` is the base class, not `BayesianRidgeEstimator`, right?

Reply:
yes, `BayesianRidgeEstimator` is only defined here for the purpose of the example